zone/trading.cpp

```diff
@@ -700,7 +700,8 @@ void Client::FinishTrade(Mob* tradingWith, bool finalizer, void* event_entry)
 						{
 							auto loot_drop_entry = LootdropEntriesRepository::NewNpcEntity();
 							loot_drop_entry.item_charges = static_cast<int8>(inst->GetCharges());
-							npc->AddLootDrop(item, loot_drop_entry, true, true);
+							bool pet = true; // setting pet(traded) flag to prevent self-found from looting this like a normal drop
+							npc->AddLootDrop(item, loot_drop_entry, true, true, false, pet, false, inst->GetCustomData());
```

Comment:
This section of code looked like a possible existing exploit. I wasn't sure if the loot was meant to be flagged as Quest or not (a comment slightly futher up reads:
```
// Add items to loottable without equipping and mark as quest.
```
But here it was just adding the loot as an actual loot drop that anyone could loot (even SF). I marked it as pet which will just make the SF checks happen properly, for now. This seems like a super edge case though, requires very specifically flagged NPCs to do it on.

Reply:
That looks about right and a proper fix. We check if the item was handed to a charmed pet owned by a ssf player at some point. I'll probably run this at least once locally before resolving it.